File: Programmers/가운데글자가져오기.cpp

#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    int strlen = s.length();
    int mid = strlen / 2;
    if(strlen % 2 == 0){
        // if length is even number
        answer += s[mid - 1];
        answer += s[mid];
        
    } else{
        // if length is odd number
        answer = s[mid];
    }
    
    return answer;
}

File: Programmers/가운데글자가져오기.swift

func solution(_ s: String) -> String {
    var answer = ""
    let arr = s.map { String($0) }
    if arr.count % 2 == 0 {
        answer.append(arr[arr.count/2 - 1])
        answer.append(arr[arr.count/2])
    } else {
        answer = arr[arr.count/2]
    }
    
    return answer
}
